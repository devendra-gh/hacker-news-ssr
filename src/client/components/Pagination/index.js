/* eslint-disable jsx-a11y/no-interactive-element-to-noninteractive-role */
import React from 'react';
import PropTypes from 'prop-types';
import { constant } from '../../constant';

const Pagination = (props) => {
  const { pageNumber, isLoading, onClickPrevNext } = props;
  return (
    <div className="carousel-container" style={{ paddingTop: '20px', textAlign: 'right' }}>
      <button
        role="navigation"
        aria-label="Previous Button"
        disabled={pageNumber === 0 || isLoading}
        name={constant.PREV}
        onClick={onClickPrevNext}
        style={{ marginRight: '10px' }}
        type="button"
        className="btn red darken-4"
      >
        Previous
      </button>
      <button
        role="navigation"
        aria-label="Next Button"
        disabled={isLoading}
        name={constant.NEXT}
        onClick={onClickPrevNext}
        type="button"
        className="btn red darken-4"
      >
        Next
      </button>
    </div>
  );
};

Pagination.propTypes = {
  pageNumber: PropTypes.number,
  isLoading: PropTypes.bool,
  onClickPrevNext: PropTypes.func,
};

Pagination.defaultProps = {
  pageNumber: 0,
  isLoading: false,
  onClickPrevNext: null,
};

export default Pagination;
