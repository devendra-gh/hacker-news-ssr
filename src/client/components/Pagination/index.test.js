import React from 'react';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Pagination from './index';

describe('<Pagination />', () => {
  let props;

  beforeEach(() => {
    props = {
      pageNumber: 0,
      isLoading: false,
      onClickPrevNext: jest.fn(),
    };
  });

  afterEach(cleanup);

  test('Renders <Pagination /> component correctly', async () => {
    const { container, getByText } = render(<Pagination />);
    expect(container).toMatchSnapshot();

    const nextButton = await waitFor(() => getByText('Next'));
    fireEvent.click(nextButton);

    const prevButton = await waitFor(() => getByText('Previous'));
    fireEvent.click(prevButton);
  });
});
