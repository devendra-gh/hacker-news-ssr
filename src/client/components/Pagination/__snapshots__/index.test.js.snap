// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Renders <Pagination /> component correctly 1`] = `
<div>
  <div
    class="carousel-container"
    style="padding-top: 20px; text-align: right;"
  >
    <button
      aria-label="Previous Button"
      class="btn red darken-4"
      disabled=""
      name="prev"
      role="navigation"
      style="margin-right: 10px;"
      type="button"
    >
      Previous
    </button>
    <button
      aria-label="Next Button"
      class="btn red darken-4"
      name="next"
      role="navigation"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
